Description: Parrot profile patches.
 firejail (0.9.58-1parrot6) UNRELEASED; urgency=medium
 .
   * Add /etc/alternatives support on most of the profiles.
   * Merge recent fixes into parrot-patches.patch.
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2019-02-16

--- firejail-0.9.58.orig/etc/QOwnNotes.profile
+++ firejail-0.9.58/etc/QOwnNotes.profile
@@ -15,7 +15,12 @@ mkdir ${DOCUMENTS}
 mkdir ${HOME}/Nextcloud/Notes
 mkdir ${HOME}.config/PBE
 mkdir ${HOME}/.local/share/PBE
+whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
 whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/Nextcloud/Notes
 whitelist ${HOME}/.config/PBE
 whitelist ${HOME}/.local/share/PBE
@@ -49,7 +54,7 @@ tracelog
 disable-mnt
 private-bin QOwnNotes,gio
 private-dev
-private-etc fonts,ld.so.cache,pulse,resolv.conf,hosts,nsswitch.conf,host.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,ld.so.cache,pulse,resolv.conf,hosts,nsswitch.conf,host.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/Viber.profile
+++ firejail-0.9.58/etc/Viber.profile
@@ -32,7 +32,7 @@ shell none
 
 disable-mnt
 private-bin sh,bash,dig,awk,Viber
-private-etc hosts,fonts,mailcap,resolv.conf,X11,pulse,alternatives,localtime,nsswitch.conf,ssl,proxychains.conf,pki,ca-certificates,crypto-policies,machine-id,asound.conf
+private-etc hosts,fonts,mailcap,resolv.conf,X11,pulse,alternatives,localtime,nsswitch.conf,ssl,proxychains.conf,pki,ca-certificates,crypto-policies,machine-id,asound.conf,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/Xvfb.profile
+++ firejail-0.9.58/etc/Xvfb.profile
@@ -41,5 +41,5 @@ private
 # private-bin Xvfb,sh,xkbcomp
 # private-bin Xvfb,sh,xkbcomp,strace,bash,cat,ls
 private-dev
-private-etc ld.so.conf,ld.so.cache,resolv.conf,host.conf,nsswitch.conf,gai.conf,hosts,hostname
+private-etc ld.so.conf,ld.so.cache,resolv.conf,host.conf,nsswitch.conf,gai.conf,hosts,hostname,alternatives
 private-tmp
--- firejail-0.9.58.orig/etc/amarok.profile
+++ firejail-0.9.58/etc/amarok.profile
@@ -7,6 +7,9 @@ include amarok.local
 include globals.local
 
 noblacklist ${MUSIC}
+noblacklist ${DOWNLOADS}
+noblacklist ${DESKTOP}
+noblacklist ${DOCUMENTS}
 
 include disable-common.inc
 include disable-devel.inc
--- firejail-0.9.58.orig/etc/ardour5.profile
+++ firejail-0.9.58/etc/ardour5.profile
@@ -11,6 +11,7 @@ noblacklist ${HOME}/.lv2
 noblacklist ${HOME}/.vst
 noblacklist ${DOCUMENTS}
 noblacklist ${MUSIC}
+noblacklist ${DOWNLOADS}
 
 include disable-common.inc
 include disable-devel.inc
--- firejail-0.9.58.orig/etc/arduino.profile
+++ firejail-0.9.58/etc/arduino.profile
@@ -10,6 +10,8 @@ noblacklist ${HOME}/.arduino15
 noblacklist ${HOME}/.java
 noblacklist ${HOME}/Arduino
 noblacklist ${DOCUMENTS}
+noblacklist ${DOWNLOADS}
+noblacklist ${DDESKTOP}
 
 # Allow access to java
 noblacklist ${PATH}/java
--- firejail-0.9.58.orig/etc/arm.profile
+++ firejail-0.9.58/etc/arm.profile
@@ -44,7 +44,7 @@ tracelog
 disable-mnt
 private-bin arm,tor,sh,bash,python*,ps,lsof,ldconfig
 private-dev
-private-etc tor,passwd,ca-certificates,ssl,pki,crypto-policies
+private-etc tor,passwd,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/atom.profile
+++ firejail-0.9.58/etc/atom.profile
@@ -37,3 +37,5 @@ private-tmp
 
 noexec ${HOME}
 noexec /tmp
+
+join-or-start atom
--- firejail-0.9.58.orig/etc/atril.profile
+++ firejail-0.9.58/etc/atril.profile
@@ -9,6 +9,8 @@ include globals.local
 noblacklist ${HOME}/.cache/atril
 noblacklist ${HOME}/.config/atril
 noblacklist ${DOCUMENTS}
+noblacklist ${DESKTOP}
+noblacklist ${DOWNLOADS}
 
 #noblacklist ${HOME}/.local/share
 # it seems to use only ${HOME}/.local/share/webkitgtk
@@ -39,9 +41,9 @@ seccomp
 shell none
 tracelog
 
-private-bin atril, atril-previewer, atril-thumbnailer
+#private-bin atril, atril-previewer, atril-thumbnailer
 private-dev
-private-etc fonts,ld.so.cache
+private-etc fonts,ld.so.cache,alternatives
 # atril uses webkit gtk to display epub files
 # waiting for globbing support in private-lib; for now hardcoding it to webkit2gtk-4.0
 #private-lib webkit2gtk-4.0 - problems on Arch with the new version of WebKit
--- firejail-0.9.58.orig/etc/authenticator.profile
+++ firejail-0.9.58/etc/authenticator.profile
@@ -40,7 +40,7 @@ disable-mnt
 # private-bin authenticator
 private-cache
 private-dev
-private-etc fonts,ld.so.cache
+private-etc fonts,ld.so.cache,alternatives
 # private-lib
 private-tmp
 
--- firejail-0.9.58.orig/etc/bibletime.profile
+++ firejail-0.9.58/etc/bibletime.profile
@@ -44,5 +44,5 @@ shell none
 
 # private-bin bibletime,qt5ct
 private-dev
-private-etc fonts,resolv.conf,sword,sword.conf,passwd,machine-id,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,resolv.conf,sword,sword.conf,passwd,machine-id,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-tmp
--- firejail-0.9.58.orig/etc/bleachbit.profile
+++ firejail-0.9.58/etc/bleachbit.profile
@@ -12,10 +12,10 @@ noblacklist ${PATH}/python3*
 noblacklist /usr/lib/python2*
 noblacklist /usr/lib/python3*
 
-include disable-common.inc
-include disable-devel.inc
-include disable-interpreters.inc
-include disable-passwdmgr.inc
+#include disable-common.inc
+#include disable-devel.inc
+#include disable-interpreters.inc
+#include disable-passwdmgr.inc
 # include disable-programs.inc
 
 caps.drop all
--- firejail-0.9.58.orig/etc/blender.profile
+++ firejail-0.9.58/etc/blender.profile
@@ -42,3 +42,5 @@ private-tmp
 
 noexec ${HOME}
 noexec /tmp
+
+join-or-start blender
--- firejail-0.9.58.orig/etc/caja.profile
+++ firejail-0.9.58/etc/caja.profile
@@ -12,6 +12,7 @@ include globals.local
 noblacklist ${HOME}/.local/share/Trash
 # noblacklist ${HOME}/.config/caja - disable-programs.inc is disabled, see below
 # noblacklist ${HOME}/.local/share/caja-python
+noblacklist ${HOME}
 
 # Allow python (blacklisted by disable-interpreters.inc)
 noblacklist ${PATH}/python2*
@@ -19,10 +20,10 @@ noblacklist ${PATH}/python3*
 noblacklist /usr/lib/python2*
 noblacklist /usr/lib/python3*
 
-include disable-common.inc
-include disable-devel.inc
-include disable-interpreters.inc
-include disable-passwdmgr.inc
+#include disable-common.inc
+#include disable-devel.inc
+#include disable-interpreters.inc
+#include disable-passwdmgr.inc
 # include disable-programs.inc
 
 caps.drop all
@@ -30,7 +31,6 @@ netfilter
 nodvd
 nogroups
 nonewprivs
-noroot
 notv
 novideo
 protocol unix
--- firejail-0.9.58.orig/etc/calligra.profile
+++ firejail-0.9.58/etc/calligra.profile
@@ -27,7 +27,7 @@ protocol unix
 seccomp
 shell none
 
-private-bin calligra,calligraauthor,calligraconverter,calligraflow,calligraplan,calligraplanwork,calligrasheets,calligrastage,calligrawords,dbus-launch,kbuildsycoca4,kdeinit4
+private-bin calligra,calligraauthor,calligraconverter,calligraflow,calligraplan,calligraplanwork,calligrasheets,calligrastage,calligrawords,dbus-launch,kbuildsycoca4,kdeinit4,alternatives
 private-dev
 
 # noexec ${HOME}
--- firejail-0.9.58.orig/etc/chromium-common.profile
+++ firejail-0.9.58/etc/chromium-common.profile
@@ -15,6 +15,11 @@ include disable-programs.inc
 
 mkdir ${HOME}/.pki
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.pki
 include whitelist-common.inc
 include whitelist-var-common.inc
--- firejail-0.9.58.orig/etc/cmus.profile
+++ firejail-0.9.58/etc/cmus.profile
@@ -27,4 +27,4 @@ seccomp
 shell none
 
 private-bin cmus
-private-etc group,machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc group,machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
--- firejail-0.9.58.orig/etc/code.profile
+++ firejail-0.9.58/etc/code.profile
@@ -36,3 +36,5 @@ private-tmp
 # in VS Code
 # noexec ${HOME}
 noexec /tmp
+
+join-or-start code
--- firejail-0.9.58.orig/etc/curl.profile
+++ firejail-0.9.58/etc/curl.profile
@@ -21,7 +21,7 @@ no3d
 nodvd
 nogroups
 nonewprivs
-noroot
+# noroot
 nosound
 notv
 nou2f
--- firejail-0.9.58.orig/etc/darktable.profile
+++ firejail-0.9.58/etc/darktable.profile
@@ -8,7 +8,10 @@ include globals.local
 
 noblacklist ${HOME}/.cache/darktable
 noblacklist ${HOME}/.config/darktable
-noblacklist ${PICTURES}
+whitelist ${DOWNLOADS}
+whitelist ${DESKTOP}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 
 include disable-common.inc
 include disable-devel.inc
--- firejail-0.9.58.orig/etc/deluge.profile
+++ firejail-0.9.58/etc/deluge.profile
@@ -21,7 +21,12 @@ include disable-passwdmgr.inc
 include disable-programs.inc
 
 mkdir ${HOME}/.config/deluge
-whitelist  ${DOWNLOADS}
+whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.config/deluge
 include whitelist-common.inc
 include whitelist-var-common.inc
--- firejail-0.9.58.orig/etc/digikam.profile
+++ firejail-0.9.58/etc/digikam.profile
@@ -37,7 +37,7 @@ shell none
 
 # private-bin program
 # private-dev - prevents libdc1394 loading; this lib is used to connect to a camera device
-# private-etc ca-certificates,ssl,pki,crypto-policies
+# private-etc ca-certificates,ssl,pki,crypto-policies,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/dillo.profile
+++ firejail-0.9.58/etc/dillo.profile
@@ -17,6 +17,11 @@ include disable-programs.inc
 mkdir ${HOME}/.dillo
 mkdir ${HOME}/.fltk
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.dillo
 whitelist ${HOME}/.fltk
 include whitelist-common.inc
--- firejail-0.9.58.orig/etc/discord-common.profile
+++ firejail-0.9.58/etc/discord-common.profile
@@ -12,6 +12,11 @@ include disable-passwdmgr.inc
 include disable-programs.inc
 
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 
 caps.drop all
 netfilter
@@ -25,9 +30,9 @@ novideo
 protocol unix,inet,inet6,netlink
 seccomp
 
-private-bin sh,xdg-mime,tr,sed,echo,head,cut,xdg-open,grep,egrep,bash,zsh
+private-bin sh,xdg-mime,tr,sed,echo,head,cut,xdg-open,grep,egrep,bash,zsh,alternatives
 private-dev
-private-etc fonts,machine-id,localtime,ld.so.cache,ca-certificates,ssl,pki,crypto-policies,resolv.conf
+private-etc fonts,machine-id,localtime,ld.so.cache,ca-certificates,ssl,pki,crypto-policies,resolv.conf,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/electron.profile
+++ firejail-0.9.58/etc/electron.profile
@@ -11,6 +11,11 @@ include disable-passwdmgr.inc
 include disable-programs.inc
 
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 
 apparmor
 caps.drop all
--- firejail-0.9.58.orig/etc/electrum.profile
+++ firejail-0.9.58/etc/electrum.profile
@@ -47,7 +47,7 @@ disable-mnt
 private-bin electrum,python*
 private-cache
 private-dev
-private-etc fonts,dconf,ca-certificates,ssl,pki,crypto-policies,machine-id
+private-etc fonts,dconf,ca-certificates,ssl,pki,crypto-policies,machine-id,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/eog.profile
+++ firejail-0.9.58/etc/eog.profile
@@ -39,7 +39,7 @@ shell none
 private-bin eog
 private-cache
 private-dev
-private-etc fonts
+private-etc fonts,alternatives
 private-lib gdk-pixbuf-2.*,gio,girepository-1.*,gvfs,libgconf-2.so.*
 private-tmp
 
--- firejail-0.9.58.orig/etc/eom.profile
+++ firejail-0.9.58/etc/eom.profile
@@ -39,7 +39,7 @@ tracelog
 
 private-bin eom
 private-dev
-private-etc fonts
+private-etc fonts,alternatives
 private-lib
 private-tmp
 
--- firejail-0.9.58.orig/etc/evince.profile
+++ firejail-0.9.58/etc/evince.profile
@@ -37,9 +37,9 @@ seccomp
 shell none
 tracelog
 
-private-bin evince,evince-previewer,evince-thumbnailer
+private-bin evince,evince-previewer,evince-thumbnailer,alternatives
 private-dev
-private-etc fonts,machine-id
+private-etc fonts,machine-id,alternatives
 
 private-lib evince,gdk-pixbuf-2.*,gio,gvfs/libgvfscommon.so,libdjvulibre.so.*,libgconf-2.so.*,libpoppler-glib.so.*,librsvg-2.so.*,gconv
 
--- firejail-0.9.58.orig/etc/falkon.profile
+++ firejail-0.9.58/etc/falkon.profile
@@ -16,6 +16,11 @@ include disable-passwdmgr.inc
 include disable-programs.inc
 
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.cache/falkon
 whitelist ${HOME}/.config/falkon
 include whitelist-common.inc
--- firejail-0.9.58.orig/etc/firefox-common.profile
+++ firejail-0.9.58/etc/firefox-common.profile
@@ -18,6 +18,11 @@ include disable-programs.inc
 
 mkdir ${HOME}/.pki
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.pki
 include whitelist-common.inc
 include whitelist-var-common.inc
@@ -32,24 +37,24 @@ netfilter
 # Also breaks enigmail apparently?
 # During a stream on Plasma it prevents the mechanism to temporarily bypass the power management, i.e. to keep the screen on
 # Therefore disable if you use that
-nodbus
+# nodbus
 nodvd
 nogroups
 nonewprivs
 noroot
 notv
-?BROWSER_DISABLE_U2F: nou2f
 protocol unix,inet,inet6,netlink
 seccomp.drop @clock,@cpu-emulation,@debug,@module,@obsolete,@raw-io,@reboot,@resources,@swap,acct,add_key,bpf,fanotify_init,io_cancel,io_destroy,io_getevents,io_setup,io_submit,ioprio_set,kcmp,keyctl,mount,name_to_handle_at,nfsservctl,ni_syscall,open_by_handle_at,personality,pivot_root,process_vm_readv,ptrace,remap_file_pages,request_key,setdomainname,sethostname,syslog,umount,umount2,userfaultfd,vhangup,vmsplice
 shell none
 #disable tracelog, it breaks or causes major issues with many firefox based browsers, see github issue #1930
 #tracelog
 
-disable-mnt
+# disable-mnt
 private-dev
 # private-etc below works fine on most distributions. There are some problems on CentOS.
-#private-etc ca-certificates,ssl,machine-id,dconf,selinux,passwd,group,hostname,hosts,localtime,nsswitch.conf,resolv.conf,xdg,gtk-2.0,gtk-3.0,X11,pango,fonts,mime.types,mailcap,asound.conf,pulse,pki,crypto-policies,ld.so.cache
-private-tmp
+private-etc ca-certificates,ssl,machine-id,dconf,selinux,passwd,group,hostname,hosts,localtime,nsswitch.conf,resolv.conf,xdg,gtk-2.0,gtk-3.0,X11,pango,fonts,mime.types,mailcap,asound.conf,pulse,pki,crypto-policies,ld.so.cache,alternatives
+#private-tmp
 
-noexec ${HOME}
+#it breaks widevinecdm module
+#noexec ${HOME}
 noexec /tmp
--- firejail-0.9.58.orig/etc/firefox.profile
+++ firejail-0.9.58/etc/firefox.profile
@@ -21,3 +21,5 @@ whitelist ${HOME}/.mozilla
 
 # Redirect
 include firefox-common.profile
+
+join-or-start firefox
--- firejail-0.9.58.orig/etc/firejail.config
+++ firejail-0.9.58/etc/firejail.config
@@ -6,7 +6,7 @@
 # apparmor yes
 
 # Disable U2F in browsers, default enabled.
-# browser-disable-u2f yes
+browser-disable-u2f no
 
 # Number of ARP probes sent when assigning an IP address for --net option,
 # default 2. This is a partial implementation of RFC 5227. A 0.5 seconds
--- firejail-0.9.58.orig/etc/flameshot.profile
+++ firejail-0.9.58/etc/flameshot.profile
@@ -20,6 +20,7 @@ ipc-namespace
 netfilter
 no3d
 # nodbus
+ignore nodbus
 nodvd
 nogroups
 nonewprivs
@@ -32,10 +33,10 @@ protocol unix,inet,inet6
 seccomp
 shell none
 
-disable-mnt
+#disable-mnt
 private-bin flameshot
 private-cache
-private-etc fonts,ld.so.conf,resolv.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,ld.so.conf,resolv.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-dev
 private-tmp
 
--- firejail-0.9.58.orig/etc/freecad.profile
+++ firejail-0.9.58/etc/freecad.profile
@@ -32,7 +32,7 @@ protocol unix
 seccomp
 shell none
 
-private-bin freecad,freecadcmd
+private-bin freecad,freecadcmd,alternatives
 private-cache
 private-dev
 private-tmp
--- firejail-0.9.58.orig/etc/gajim.profile
+++ firejail-0.9.58/etc/gajim.profile
@@ -45,9 +45,9 @@ shell none
 tracelog
 
 disable-mnt
-private-bin python,python3,sh,gpg,gpg2,gajim,bash,zsh,paplay,gajim-history-manager
+private-bin python,python3,sh,gpg,gpg2,gajim,bash,zsh,paplay,gajim-history-manager,alternatives
 private-dev
-private-etc alsa,asound.conf,ca-certificates,crypto-policies,fonts,group,hostname,hosts,ld.so.cache,ld.so.conf,localtime,machine-id,passwd,pki,pulse,resolv.conf,ssl
+private-etc alsa,asound.conf,ca-certificates,crypto-policies,fonts,group,hostname,hosts,ld.so.cache,ld.so.conf,localtime,machine-id,passwd,pki,pulse,resolv.conf,ssl,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/galculator.profile
+++ firejail-0.9.58/etc/galculator.profile
@@ -38,6 +38,6 @@ tracelog
 
 private-bin galculator
 private-dev
-private-etc fonts
+private-etc fonts,alternatives
 private-lib
 private-tmp
--- firejail-0.9.58.orig/etc/gcloud.profile
+++ firejail-0.9.58/etc/gcloud.profile
@@ -32,7 +32,7 @@ tracelog
 
 disable-mnt
 private-dev
-private-etc ca-certificates,ssl,hosts,localtime,nsswitch.conf,resolv.conf,pki,crypto-policies,ld.so.cache
+private-etc ca-certificates,ssl,hosts,localtime,nsswitch.conf,resolv.conf,pki,crypto-policies,ld.so.cache,alternatives
 private-tmp
 
 noexec /tmp
--- firejail-0.9.58.orig/etc/geany.profile
+++ firejail-0.9.58/etc/geany.profile
@@ -8,9 +8,9 @@ include globals.local
 
 noblacklist ${HOME}/.config/geany
 
-include disable-common.inc
+#include disable-common.inc
 include disable-passwdmgr.inc
-include disable-programs.inc
+#include disable-programs.inc
 
 caps.drop all
 netfilter
@@ -18,7 +18,7 @@ no3d
 nodvd
 nogroups
 nonewprivs
-noroot
+#noroot
 nosound
 notv
 nou2f
--- firejail-0.9.58.orig/etc/gedit.profile
+++ firejail-0.9.58/etc/gedit.profile
@@ -10,11 +10,11 @@ noblacklist ${HOME}/.config/enchant
 noblacklist ${HOME}/.config/gedit
 noblacklist ${HOME}/.gitconfig
 
-include disable-common.inc
+#include disable-common.inc
 # include disable-devel.inc
 # include disable-interpreters.inc
 include disable-passwdmgr.inc
-include disable-programs.inc
+#include disable-programs.inc
 
 include whitelist-var-common.inc
 
@@ -27,7 +27,7 @@ no3d
 nodvd
 nogroups
 nonewprivs
-noroot
+#noroot
 nosound
 notv
 nou2f
--- firejail-0.9.58.orig/etc/ghostwriter.profile
+++ firejail-0.9.58/etc/ghostwriter.profile
@@ -52,7 +52,7 @@ tracelog
 #private-bin ghostwriter,pandoc
 private-cache
 private-dev
-private-etc cups,crypto-policies,localtime,drirc,fonts,gtk-3.0,dconf,machine-id
+private-etc cups,crypto-policies,localtime,drirc,fonts,gtk-3.0,dconf,machine-id,alternatives
 # Breaks Translation
 #private-lib
 private-tmp
--- firejail-0.9.58.orig/etc/gimp.profile
+++ firejail-0.9.58/etc/gimp.profile
@@ -40,3 +40,5 @@ private-tmp
 # if you are not using external plugins, you can enable noexec statement below
 # noexec ${HOME}
 noexec /tmp
+
+join-or-start gimp
--- firejail-0.9.58.orig/etc/gitter.profile
+++ firejail-0.9.58/etc/gitter.profile
@@ -15,6 +15,11 @@ include disable-passwdmgr.inc
 include disable-programs.inc
 
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.config/autostart
 whitelist ${HOME}/.config/Gitter
 include whitelist-var-common.inc
@@ -35,7 +40,7 @@ shell none
 
 disable-mnt
 private-bin bash,env,gitter
-private-etc fonts,pulse,resolv.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,pulse,resolv.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-opt Gitter
 private-dev
 private-tmp
--- firejail-0.9.58.orig/etc/gnome-chess.profile
+++ firejail-0.9.58/etc/gnome-chess.profile
@@ -35,7 +35,7 @@ tracelog
 disable-mnt
 private-bin fairymax,gnome-chess,hoichess
 private-dev
-private-etc fonts,gnome-chess
+private-etc fonts,gnome-chess,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/gnome-logs.profile
+++ firejail-0.9.58/etc/gnome-logs.profile
@@ -37,7 +37,7 @@ shell none
 disable-mnt
 private-bin gnome-logs
 private-dev
-private-etc fonts,localtime,machine-id
+private-etc fonts,localtime,machine-id,alternatives
 private-lib gdk-pixbuf-2.*,gio,gvfs/libgvfscommon.so,libgconf-2.so.*,librsvg-2.so.*
 private-tmp
 writable-var-log
--- firejail-0.9.58.orig/etc/gnome-music.profile
+++ firejail-0.9.58/etc/gnome-music.profile
@@ -40,7 +40,7 @@ tracelog
 
 private-bin gnome-music,python*,env,gio-launch-desktop,yelp
 private-dev
-private-etc fonts,machine-id,pulse,asound.conf
+private-etc fonts,machine-id,pulse,asound.conf,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/gnome-pie.profile
+++ firejail-0.9.58/etc/gnome-pie.profile
@@ -34,7 +34,7 @@ shell none
 disable-mnt
 private-cache
 private-dev
-private-etc fonts
+private-etc fonts,alternatives
 private-lib gdk-pixbuf-2.*,gio,gvfs/libgvfscommon.so,libgconf-2.so.*,librsvg-2.so.*
 private-tmp
 
--- firejail-0.9.58.orig/etc/gnome-recipes.profile
+++ firejail-0.9.58/etc/gnome-recipes.profile
@@ -38,7 +38,7 @@ shell none
 disable-mnt
 private-bin gnome-recipes,tar
 private-dev
-private-etc ca-certificates,fonts,ssl,crypto-policies,pki
+private-etc ca-certificates,fonts,ssl,crypto-policies,pki,alternatives
 # private-lib works for me with Gnome Shell 3.26.2, Mutter WM (Arch Linux)
 # not widely tested though, leaving it to devs discretion to enable it later
 #private-lib gdk-pixbuf-2.0,gio,gvfs/libgvfscommon.so,libgconf-2.so.4,libgnutls.so.30,libjpeg.so.8,libp11-kit.so.0,libproxy.so.1,librsvg-2.so.2
--- firejail-0.9.58.orig/etc/goobox.profile
+++ firejail-0.9.58/etc/goobox.profile
@@ -31,5 +31,5 @@ tracelog
 
 # private-bin goobox
 private-dev
-# private-etc fonts,machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies
+# private-etc fonts,machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 # private-tmp
--- firejail-0.9.58.orig/etc/gpredict.profile
+++ firejail-0.9.58/etc/gpredict.profile
@@ -33,7 +33,7 @@ tracelog
 
 private-bin gpredict
 private-dev
-private-etc fonts,resolv.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,resolv.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/gradio.profile
+++ firejail-0.9.58/etc/gradio.profile
@@ -34,7 +34,7 @@ protocol unix,inet,inet6
 seccomp
 shell none
 
-private-etc asound.conf,ca-certificates,fonts,host.conf,hostname,hosts,pulse,resolv.conf,ssl,pki,crypto-policies,gtk-3.0,xdg,machine-id
+private-etc asound.conf,ca-certificates,fonts,host.conf,hostname,hosts,pulse,resolv.conf,ssl,pki,crypto-policies,gtk-3.0,xdg,machine-id,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/gwenview.profile
+++ firejail-0.9.58/etc/gwenview.profile
@@ -43,7 +43,7 @@ shell none
 
 private-bin gwenview,gimp*,kbuildsycoca4,kdeinit4
 private-dev
-private-etc fonts,gimp,gtk-2.0,kde4rc,kde5rc,ld.so.cache,machine-id,pulse,xdg
+private-etc fonts,gimp,gtk-2.0,kde4rc,kde5rc,ld.so.cache,machine-id,pulse,xdg,alternatives
 
 # memory-deny-write-execute
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/kdenlive.profile
+++ firejail-0.9.58/etc/kdenlive.profile
@@ -32,7 +32,7 @@ shell none
 
 private-bin kdenlive,kdenlive_render,dbus-launch,melt,ffmpeg,ffplay,ffprobe,dvdauthor,genisoimage,vlc,xine,kdeinit5,kshell5,kdeinit5_shutdown,kdeinit5_wrapper,kdeinit4,kshell4,kdeinit4_shutdown,kdeinit4_wrapper,mlt-melt
 private-dev
-# private-etc alternatives,fonts,kde4rc,kde5rc,ld.so.cache,machine-id,passwd,pulse,xdg,X11
+# private-etc alternatives,fonts,kde4rc,kde5rc,ld.so.cache,machine-id,passwd,pulse,xdg,X11,alternatives
 
 # noexec ${HOME}
 noexec /tmp
--- firejail-0.9.58.orig/etc/keepassx.profile
+++ firejail-0.9.58/etc/keepassx.profile
@@ -41,7 +41,7 @@ tracelog
 
 private-bin keepassx,keepassx2
 private-dev
-private-etc fonts,machine-id
+private-etc fonts,machine-id,alternatives
 private-tmp
 
 memory-deny-write-execute
--- firejail-0.9.58.orig/etc/keepassxc.profile
+++ firejail-0.9.58/etc/keepassxc.profile
@@ -34,7 +34,6 @@ nonewprivs
 noroot
 nosound
 notv
-nou2f
 novideo
 protocol unix
 seccomp
@@ -42,7 +41,7 @@ shell none
 
 private-bin keepassxc
 private-dev
-private-etc fonts,ld.so.cache,machine-id
+private-etc fonts,ld.so.cache,machine-id,alternatives
 private-tmp
 
 # 2.2.4 crashes on database open
--- firejail-0.9.58.orig/etc/kwin_x11.profile
+++ firejail-0.9.58/etc/kwin_x11.profile
@@ -36,7 +36,7 @@ tracelog
 disable-mnt
 private-bin kwin_x11
 private-dev
-private-etc drirc,fonts,kde5rc,ld.so.cache,machine-id,xdg
+private-etc drirc,fonts,kde5rc,ld.so.cache,machine-id,xdg,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/kwrite.profile
+++ firejail-0.9.58/etc/kwrite.profile
@@ -44,7 +44,7 @@ tracelog
 
 private-bin kwrite,kbuildsycoca4,kdeinit4
 private-dev
-private-etc fonts,kde4rc,kde5rc,ld.so.cache,machine-id,pulse,xdg
+private-etc fonts,kde4rc,kde5rc,ld.so.cache,machine-id,pulse,xdg,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/leafpad.profile
+++ firejail-0.9.58/etc/leafpad.profile
@@ -22,7 +22,6 @@ no3d
 nodvd
 nogroups
 nonewprivs
-noroot
 nosound
 notv
 nou2f
--- firejail-0.9.58.orig/etc/lollypop.profile
+++ firejail-0.9.58/etc/lollypop.profile
@@ -38,7 +38,7 @@ seccomp
 shell none
 
 private-dev
-private-etc asound.conf,ca-certificates,fonts,host.conf,hostname,hosts,pulse,resolv.conf,ssl,pki,crypto-policies,gtk-3.0,xdg,machine-id
+private-etc asound.conf,ca-certificates,fonts,host.conf,hostname,hosts,pulse,resolv.conf,ssl,pki,crypto-policies,gtk-3.0,xdg,machine-id,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/masterpdfeditor.profile
+++ firejail-0.9.58/etc/masterpdfeditor.profile
@@ -41,7 +41,7 @@ tracelog
 private-bin masterpdfeditor*
 private-cache
 private-dev
-private-etc fonts
+private-etc fonts,alternatives
 # private-lib
 private-tmp
 
--- firejail-0.9.58.orig/etc/mate-color-select.profile
+++ firejail-0.9.58/etc/mate-color-select.profile
@@ -34,7 +34,7 @@ shell none
 
 disable-mnt
 private-bin mate-color-select
-private-etc fonts
+private-etc fonts,alternatives
 private-dev
 private-lib
 private-tmp
--- firejail-0.9.58.orig/etc/mate-dictionary.profile
+++ firejail-0.9.58/etc/mate-dictionary.profile
@@ -36,7 +36,7 @@ shell none
 
 disable-mnt
 private-bin mate-dictionary
-private-etc fonts,resolv.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,resolv.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-opt mate-dictionary
 private-dev
 private-tmp
--- firejail-0.9.58.orig/etc/midori.profile
+++ firejail-0.9.58/etc/midori.profile
@@ -24,6 +24,11 @@ mkdir ${HOME}/.local/share/webkit
 mkdir ${HOME}/.local/share/webkitgtk
 mkdir ${HOME}/.pki
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.cache/gnome-mplayer/plugin
 whitelist ${HOME}/.cache/midori
 whitelist ${HOME}/.config/gnome-mplayer
--- firejail-0.9.58.orig/etc/min.profile
+++ firejail-0.9.58/etc/min.profile
@@ -17,6 +17,11 @@ include disable-programs.inc
 
 mkdir ${HOME}/.pki
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.pki
 include whitelist-common.inc
 include whitelist-var-common.inc
@@ -43,7 +48,7 @@ disable-mnt
 private-cache
 private-dev
 # private-etc below works fine on most distributions. There are some problems on CentOS.
-private-etc ca-certificates,ssl,machine-id,dconf,selinux,passwd,group,hostname,hosts,localtime,nsswitch.conf,resolv.conf,xdg,gtk-2.0,gtk-3.0,X11,pango,fonts,mime.types,mailcap,asound.conf,pulse,pki,crypto-policies,ld.so.cache
+private-etc ca-certificates,ssl,machine-id,dconf,selinux,passwd,group,hostname,hosts,localtime,nsswitch.conf,resolv.conf,xdg,gtk-2.0,gtk-3.0,X11,pango,fonts,mime.types,mailcap,asound.conf,pulse,pki,crypto-policies,ld.so.cache,alternatives
 private-tmp
 
 # memory-deny-write-execute
--- firejail-0.9.58.orig/etc/minetest.profile
+++ firejail-0.9.58/etc/minetest.profile
@@ -38,7 +38,7 @@ disable-mnt
 private-bin minetest
 private-dev
 # private-etc needs to be updated, see #1702
-#private-etc asound.conf,ca-certificates,drirc,fonts,group,host.conf,hostname,hosts,ld.so.cache,ld.so.preload,localtime,nsswitch.conf,passwd,pulse,resolv.conf,ssl,pki,crypto-policies,machine-id
+#private-etc asound.conf,ca-certificates,drirc,fonts,group,host.conf,hostname,hosts,ld.so.cache,ld.so.preload,localtime,nsswitch.conf,passwd,pulse,resolv.conf,ssl,pki,crypto-policies,machine-id,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/ms-office.profile
+++ firejail-0.9.58/etc/ms-office.profile
@@ -37,7 +37,7 @@ tracelog
 
 disable-mnt
 private-bin bash,fonts,env,jak,ms-office,python*,sh
-private-etc resolv.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc resolv.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-dev
 private-tmp
 
--- firejail-0.9.58.orig/etc/musixmatch.profile
+++ firejail-0.9.58/etc/musixmatch.profile
@@ -31,7 +31,7 @@ seccomp
 
 disable-mnt
 private-dev
-private-etc machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 
 noexec ${HOME}
 noexec /tmp
--- firejail-0.9.58.orig/etc/ocenaudio.profile
+++ firejail-0.9.58/etc/ocenaudio.profile
@@ -43,7 +43,7 @@ tracelog
 private-bin ocenaudio
 private-cache
 private-dev
-private-etc asound.conf,fonts,ld.so.cache,pulse
+private-etc asound.conf,fonts,ld.so.cache,pulse,alternatives
 # private-lib
 private-tmp
 
--- firejail-0.9.58.orig/etc/parole.profile
+++ firejail-0.9.58/etc/parole.profile
@@ -27,4 +27,4 @@ shell none
 
 private-bin parole,dbus-launch
 private-cache
-private-etc passwd,group,fonts,machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc passwd,group,fonts,machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
--- firejail-0.9.58.orig/etc/pdfchain.profile
+++ firejail-0.9.58/etc/pdfchain.profile
@@ -34,7 +34,7 @@ shell none
 
 private-bin pdfchain,pdftk,sh
 private-dev
-private-etc dconf,fonts,gtk-3.0,xdg
+private-etc dconf,fonts,gtk-3.0,xdg,alternatives
 private-tmp
 
 memory-deny-write-execute
--- firejail-0.9.58.orig/etc/pluma.profile
+++ firejail-0.9.58/etc/pluma.profile
@@ -8,11 +8,11 @@ include globals.local
 
 noblacklist ${HOME}/.config/pluma
 
-include disable-common.inc
-include disable-devel.inc
-include disable-interpreters.inc
+#include disable-common.inc
+#include disable-devel.inc
+#include disable-interpreters.inc
 include disable-passwdmgr.inc
-include disable-programs.inc
+#include disable-programs.inc
 
 include whitelist-var-common.inc
 
@@ -25,7 +25,7 @@ no3d
 nodvd
 nogroups
 nonewprivs
-noroot
+#noroot
 nosound
 notv
 nou2f
@@ -35,12 +35,14 @@ seccomp
 shell none
 tracelog
 
-private-bin pluma
+#private-bin pluma
 private-dev
 # private-etc fonts
-private-lib pluma
-private-tmp
+#private-lib pluma
+#private-tmp
 
 memory-deny-write-execute
 noexec ${HOME}
 noexec /tmp
+
+join-or-start pluma
--- firejail-0.9.58.orig/etc/ppsspp.profile
+++ firejail-0.9.58/etc/ppsspp.profile
@@ -37,7 +37,7 @@ shell none
 
 # private-dev is disabled to allow controller support
 #private-dev
-private-etc asound.conf,ca-certificates,drirc,fonts,group,host.conf,hostname,hosts,ld.so.cache,ld.so.preload,localtime,nsswitch.conf,passwd,pulse,resolv.conf,ssl,pki,crypto-policies,machine-id
+private-etc asound.conf,ca-certificates,drirc,fonts,group,host.conf,hostname,hosts,ld.so.cache,ld.so.preload,localtime,nsswitch.conf,passwd,pulse,resolv.conf,ssl,pki,crypto-policies,machine-id,alternatives
 private-opt ppsspp
 private-tmp
 
--- firejail-0.9.58.orig/etc/pybitmessage.profile
+++ firejail-0.9.58/etc/pybitmessage.profile
@@ -42,7 +42,7 @@ shell none
 disable-mnt
 private-bin pybitmessage,python*,sh,ldconfig,env,bash,stat
 private-dev
-private-etc PyBitmessage,PyBitmessage.conf,Trolltech.conf,fonts,gtk-2.0,hosts,ld.so.cache,ld.so.preload,localtime,pki,resolv.conf,selinux,sni-qt.conf,system-fips,xdg,ca-certificates,ssl,pki,crypto-policies
+private-etc PyBitmessage,PyBitmessage.conf,Trolltech.conf,fonts,gtk-2.0,hosts,ld.so.cache,ld.so.preload,localtime,pki,resolv.conf,selinux,sni-qt.conf,system-fips,xdg,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-tmp
 
 memory-deny-write-execute
--- firejail-0.9.58.orig/etc/qbittorrent.profile
+++ firejail-0.9.58/etc/qbittorrent.profile
@@ -26,7 +26,12 @@ include disable-programs.inc
 mkdir ${HOME}/.cache/qBittorrent
 mkdir ${HOME}/.config/qBittorrent
 mkdir ${HOME}/.local/share/data/qBittorrent
-whitelist  ${DOWNLOADS}
+whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.cache/qBittorrent
 whitelist ${HOME}/.config/qBittorrent
 whitelist ${HOME}/.config/qBittorrentrc
@@ -60,3 +65,5 @@ private-tmp
 # memory-deny-write-execute - problems on  Arch, see #1690 on GitHub repo
 noexec ${HOME}
 noexec /tmp
+
+join-or-start qbittorrent
--- firejail-0.9.58.orig/etc/qtox.profile
+++ firejail-0.9.58/etc/qtox.profile
@@ -16,6 +16,11 @@ include disable-programs.inc
 
 mkdir ${HOME}/.config/tox
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.config/tox
 include whitelist-common.inc
 include whitelist-var-common.inc
@@ -36,7 +41,7 @@ tracelog
 
 disable-mnt
 private-bin qtox
-private-etc fonts,resolv.conf,ld.so.cache,localtime,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,resolv.conf,ld.so.cache,localtime,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-dev
 private-tmp
 
--- firejail-0.9.58.orig/etc/qupzilla.profile
+++ firejail-0.9.58/etc/qupzilla.profile
@@ -34,7 +34,7 @@ seccomp.drop @clock,@cpu-emulation,@debu
 # tracelog
 
 private-dev
-# private-etc passwd,group,hostname,hosts,localtime,nsswitch.conf,resolv.conf,gtk-2.0,pango,fonts,adobe,mime.types,mailcap,asound.conf,pulse,machine-id,ca-certificates,ssl,pki,crypto-policies
+# private-etc passwd,group,hostname,hosts,localtime,nsswitch.conf,resolv.conf,gtk-2.0,pango,fonts,adobe,mime.types,mailcap,asound.conf,pulse,machine-id,ca-certificates,ssl,pki,crypto-policies,alternatives
 # private-tmp - interferes with the opening of downloaded files
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/qutebrowser.profile
+++ firejail-0.9.58/etc/qutebrowser.profile
@@ -28,6 +28,11 @@ mkdir ${HOME}/.cache/qutebrowser
 mkdir ${HOME}/.config/qutebrowser
 mkdir ${HOME}/.local/share/qutebrowser
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.cache/qutebrowser
 whitelist ${HOME}/.config/qutebrowser
 whitelist ${HOME}/.local/share/qutebrowser
--- firejail-0.9.58.orig/etc/ricochet.profile
+++ firejail-0.9.58/etc/ricochet.profile
@@ -15,6 +15,9 @@ include disable-passwdmgr.inc
 include disable-programs.inc
 
 whitelist ${DOWNLOADS}
+whitelist ${DESKTOP}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.local/share/Ricochet
 include whitelist-common.inc
 
@@ -36,7 +39,7 @@ shell none
 disable-mnt
 private-bin ricochet,tor
 private-dev
-#private-etc fonts,tor,X11,alternatives,ca-certificates,ssl,pki,crypto-policies
+#private-etc fonts,tor,X11,alternatives,ca-certificates,ssl,pki,crypto-policies,alternatives
 
 noexec ${HOME}
 noexec /tmp
--- firejail-0.9.58.orig/etc/seamonkey.profile
+++ firejail-0.9.58/etc/seamonkey.profile
@@ -18,6 +18,11 @@ include disable-programs.inc
 mkdir ${HOME}/.cache/mozilla
 mkdir ${HOME}/.mozilla
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.cache/gnome-mplayer/plugin
 whitelist ${HOME}/.cache/mozilla
 whitelist ${HOME}/.config/gnome-mplayer
--- firejail-0.9.58.orig/etc/signal-desktop.profile
+++ firejail-0.9.58/etc/signal-desktop.profile
@@ -15,6 +15,11 @@ include disable-passwdmgr.inc
 
 mkdir ${HOME}/.config/Signal
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.config/Signal
 include whitelist-common.inc
 include whitelist-var-common.inc
--- firejail-0.9.58.orig/etc/slack.profile
+++ firejail-0.9.58/etc/slack.profile
@@ -37,5 +37,5 @@ shell none
 disable-mnt
 private-bin slack,locale
 private-dev
-private-etc asound.conf,ca-certificates,fonts,group,passwd,pulse,resolv.conf,ssl,ld.so.conf,ld.so.cache,localtime,pki,crypto-policies,machine-id
+private-etc asound.conf,ca-certificates,fonts,group,passwd,pulse,resolv.conf,ssl,ld.so.conf,ld.so.cache,localtime,pki,crypto-policies,machine-id,alternatives
 private-tmp
--- firejail-0.9.58.orig/etc/spotify.profile
+++ firejail-0.9.58/etc/spotify.profile
@@ -46,7 +46,7 @@ tracelog
 disable-mnt
 private-bin spotify,bash,sh,zenity
 private-dev
-private-etc fonts,group,ld.so.cache,machine-id,pulse,resolv.conf,hosts,nsswitch.conf,host.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,group,ld.so.cache,machine-id,pulse,resolv.conf,hosts,nsswitch.conf,host.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-opt spotify
 private-tmp
 
--- firejail-0.9.58.orig/etc/standardnotes-desktop.profile
+++ firejail-0.9.58/etc/standardnotes-desktop.profile
@@ -38,7 +38,7 @@ seccomp
 disable-mnt
 private-dev
 private-tmp
-private-etc ca-certificates,fonts,host.conf,hostname,hosts,resolv.conf,ssl,pki,crypto-policies,xdg
+private-etc ca-certificates,fonts,host.conf,hostname,hosts,resolv.conf,ssl,pki,crypto-policies,xdg,alternatives
 
 noexec ${HOME}
 noexec /tmp
--- firejail-0.9.58.orig/etc/start-tor-browser.profile
+++ firejail-0.9.58/etc/start-tor-browser.profile
@@ -34,7 +34,7 @@ shell none
 disable-mnt
 private-bin bash,sh,grep,tail,env,gpg,id,readlink,dirname,test,mkdir,ln,sed,cp,rm,getconf
 private-dev
-private-etc fonts,hostname,hosts,resolv.conf,pki,ssl,ca-certificates,crypto-policies,alsa,asound.conf,pulse,machine-id,ld.so.cache
+private-etc fonts,hostname,hosts,resolv.conf,pki,ssl,ca-certificates,crypto-policies,alsa,asound.conf,pulse,machine-id,ld.so.cache,alternatives
 private-tmp
 
 noexec /tmp
--- firejail-0.9.58.orig/etc/supertuxkart.profile
+++ firejail-0.9.58/etc/supertuxkart.profile
@@ -46,7 +46,7 @@ disable-mnt
 private-bin supertuxkart
 private-cache
 private-dev
-private-etc resolv.conf,ca-certificates,ssl,hosts,machine-id,xdg,openal,crypto-policies,pki,drirc,system-fips,selinux
+private-etc resolv.conf,ca-certificates,ssl,hosts,machine-id,xdg,openal,crypto-policies,pki,drirc,system-fips,selinux,alternatives
 private-tmp
 private-opt none
 private-srv none
--- firejail-0.9.58.orig/etc/surf.profile
+++ firejail-0.9.58/etc/surf.profile
@@ -32,7 +32,7 @@ tracelog
 disable-mnt
 private-bin ls,surf,sh,bash,curl,dmenu,printf,sed,sleep,st,stterm,xargs,xprop
 private-dev
-private-etc passwd,group,hosts,resolv.conf,fonts,ssl,pki,ca-certificates,crypto-policies
+private-etc passwd,group,hosts,resolv.conf,fonts,ssl,pki,ca-certificates,crypto-policies,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/terasology.profile
+++ firejail-0.9.58/etc/terasology.profile
@@ -44,7 +44,7 @@ shell none
 
 disable-mnt
 private-dev
-private-etc asound.conf,ca-certificates,dbus-1,drirc,fonts,group,gtk-2.0,gtk-3.0,host.conf,hostname,hosts,ld.so.cache,ld.so.preload,localtime,lsb-release,machine-id,mime.types,passwd,pulse,resolv.conf,ssl,java-8-openjdk,java-7-openjdk,pki,crypto-policies
+private-etc asound.conf,ca-certificates,dbus-1,drirc,fonts,group,gtk-2.0,gtk-3.0,host.conf,hostname,hosts,ld.so.cache,ld.so.preload,localtime,lsb-release,machine-id,mime.types,passwd,pulse,resolv.conf,ssl,java-8-openjdk,java-7-openjdk,pki,crypto-policies,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/thunderbird.profile
+++ firejail-0.9.58/etc/thunderbird.profile
@@ -43,5 +43,5 @@ writable-run-user
 # allow browsers
 # Redirect
 # Uncomment if you use enigmail
-# ignore nodbus
-include firefox.profile
+ignore nodbus
+include firefox-common.profile
--- firejail-0.9.58.orig/etc/torbrowser-launcher.profile
+++ firejail-0.9.58/etc/torbrowser-launcher.profile
@@ -35,13 +35,13 @@ netfilter
 nodbus
 nodvd
 nogroups
-nonewprivs
+#nonewprivs
 noroot
 notv
 nou2f
 novideo
-protocol unix,inet,inet6
-seccomp.drop @clock,@cpu-emulation,@debug,@module,@obsolete,@raw-io,@reboot,@resources,@swap,acct,add_key,bpf,fanotify_init,io_cancel,io_destroy,io_getevents,io_setup,io_submit,ioprio_set,kcmp,keyctl,mount,name_to_handle_at,nfsservctl,ni_syscall,open_by_handle_at,personality,pivot_root,process_vm_readv,ptrace,remap_file_pages,request_key,setdomainname,sethostname,syslog,umount,umount2,userfaultfd,vhangup,vmsplice
+#protocol unix,inet,inet6
+#seccomp.drop @clock,@cpu-emulation,@debug,@module,@obsolete,@raw-io,@reboot,@resources,@swap,acct,add_key,bpf,fanotify_init,io_cancel,io_destroy,io_getevents,io_setup,io_submit,ioprio_set,kcmp,keyctl,mount,name_to_handle_at,nfsservctl,ni_syscall,open_by_handle_at,personality,pivot_root,process_vm_readv,ptrace,remap_file_pages,request_key,setdomainname,sethostname,syslog,umount,umount2,userfaultfd,vhangup,vmsplice
 shell none
 # tracelog may cause issues, see github issue #1930
 #tracelog
@@ -49,7 +49,7 @@ shell none
 disable-mnt
 private-bin bash,cp,dirname,env,expr,file,getconf,gpg,grep,id,ln,mkdir,python*,readlink,rm,sed,sh,tail,tar,tclsh,test,tor-browser-en,torbrowser-launcher,xz
 private-dev
-private-etc fonts,hostname,hosts,resolv.conf,pki,ssl,ca-certificates,crypto-policies,alsa,asound.conf,pulse,machine-id,ld.so.cache
+private-etc fonts,hostname,hosts,resolv.conf,pki,ssl,ca-certificates,crypto-policies,alsa,asound.conf,pulse,machine-id,ld.so.cache,alternatives
 private-tmp
 
 noexec /tmp
--- firejail-0.9.58.orig/etc/totem.profile
+++ firejail-0.9.58/etc/totem.profile
@@ -36,7 +36,7 @@ private-bin totem
 # totem needs access to ~/.cache/tracker or it exits
 #private-cache
 private-dev
-# private-etc fonts,machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies
+# private-etc fonts,machine-id,pulse,asound.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/transmission-gtk.profile
+++ firejail-0.9.58/etc/transmission-gtk.profile
@@ -17,7 +17,12 @@ include disable-programs.inc
 
 mkdir ${HOME}/.cache/transmission
 mkdir ${HOME}/.config/transmission
-whitelist  ${DOWNLOADS}
+whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.cache/transmission
 whitelist ${HOME}/.config/transmission
 include whitelist-common.inc
--- firejail-0.9.58.orig/etc/transmission-qt.profile
+++ firejail-0.9.58/etc/transmission-qt.profile
@@ -17,7 +17,12 @@ include disable-programs.inc
 
 mkdir ${HOME}/.cache/transmission
 mkdir ${HOME}/.config/transmission
-whitelist  ${DOWNLOADS}
+whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.cache/transmission
 whitelist ${HOME}/.config/transmission
 include whitelist-common.inc
--- firejail-0.9.58.orig/etc/uget-gtk.profile
+++ firejail-0.9.58/etc/uget-gtk.profile
@@ -14,6 +14,11 @@ include disable-programs.inc
 
 mkdir ${HOME}/.config/uGet
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 whitelist ${HOME}/.config/uGet
 include whitelist-common.inc
 
--- firejail-0.9.58.orig/etc/vim.profile
+++ firejail-0.9.58/etc/vim.profile
@@ -19,7 +19,6 @@ netfilter
 nodvd
 nogroups
 nonewprivs
-noroot
 notv
 nou2f
 novideo
--- firejail-0.9.58.orig/etc/virtualbox.profile
+++ firejail-0.9.58/etc/virtualbox.profile
@@ -22,6 +22,11 @@ mkdir ${HOME}/VirtualBox VMs
 whitelist ${HOME}/.config/VirtualBox
 whitelist ${HOME}/VirtualBox VMs
 whitelist ${DOWNLOADS}
+whitelist ${MUSIC}
+whitelist ${DESKTOP}
+whitelist ${VIDEOS}
+whitelist ${DOCUMENTS}
+whitelist ${PICTURES}
 include whitelist-common.inc
 include whitelist-var-common.inc
 
--- firejail-0.9.58.orig/etc/vlc.profile
+++ firejail-0.9.58/etc/vlc.profile
@@ -41,3 +41,5 @@ private-tmp
 #memory-deny-write-execute
 noexec ${HOME}
 noexec /tmp
+
+join-or-start vlc
--- /dev/null
+++ firejail-0.9.58/etc/vscodium.profile
@@ -0,0 +1,42 @@
+# Firejail profile for Visual Studio Code
+# This file is overwritten after every install/update
+# Persistent local customizations
+include code.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.vscode
+noblacklist ${HOME}/.vscode-oss
+noblacklist ${HOME}/.config/Code
+noblacklist ${HOME}/.VSCodium
+
+include disable-common.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+
+caps.drop all
+net none
+netfilter
+nodvd
+nogroups
+nonewprivs
+noroot
+nosound
+notv
+nou2f
+novideo
+protocol unix,inet,inet6,netlink
+seccomp
+shell none
+
+private-cache
+private-dev
+private-tmp
+
+# Disabling noexec ${HOME} for now since it will
+# probably interfere with running some programmes
+# in VS Code
+# noexec ${HOME}
+noexec /tmp
+
+join-or-start vscodium
--- firejail-0.9.58.orig/etc/w3m.profile
+++ firejail-0.9.58/etc/w3m.profile
@@ -36,5 +36,5 @@ tracelog
 # private-bin w3m
 private-cache
 private-dev
-private-etc resolv.conf,ssl,pki,ca-certificates,crypto-policies
+private-etc resolv.conf,ssl,pki,ca-certificates,crypto-policies,alternatives
 private-tmp
--- firejail-0.9.58.orig/etc/wget.profile
+++ firejail-0.9.58/etc/wget.profile
@@ -23,7 +23,6 @@ no3d
 nodvd
 nogroups
 nonewprivs
-noroot
 nosound
 notv
 nou2f
--- firejail-0.9.58.orig/etc/whois.profile
+++ firejail-0.9.58/etc/whois.profile
@@ -24,7 +24,6 @@ nodbus
 nodvd
 nogroups
 nonewprivs
-noroot
 nosound
 notv
 nou2f
--- firejail-0.9.58.orig/etc/wire-desktop.profile
+++ firejail-0.9.58/etc/wire-desktop.profile
@@ -37,5 +37,5 @@ shell none
 disable-mnt
 private-bin wire-desktop
 private-dev
-private-etc fonts,machine-id,resolv.conf,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,machine-id,resolv.conf,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-tmp
--- firejail-0.9.58.orig/etc/x-terminal-emulator.profile
+++ firejail-0.9.58/etc/x-terminal-emulator.profile
@@ -11,7 +11,6 @@ net none
 netfilter
 nodbus
 nogroups
-noroot
 nou2f
 protocol unix
 seccomp
--- firejail-0.9.58.orig/etc/xiphos.profile
+++ firejail-0.9.58/etc/xiphos.profile
@@ -38,5 +38,5 @@ tracelog
 
 private-bin xiphos
 private-dev
-private-etc fonts,resolv.conf,sword,ca-certificates,ssl,pki,crypto-policies
+private-etc fonts,resolv.conf,sword,ca-certificates,ssl,pki,crypto-policies,alternatives
 private-tmp
--- firejail-0.9.58.orig/etc/xonotic.profile
+++ firejail-0.9.58/etc/xonotic.profile
@@ -36,7 +36,7 @@ shell none
 disable-mnt
 private-bin bash,blind-id,darkplaces-glx,darkplaces-sdl,dirname,grep,ldd,netstat,ps,readlink,sh,uname,xonotic,xonotic-glx,xonotic-linux32-dedicated,xonotic-linux32-glx,xonotic-linux32-sdl,xonotic-linux64-dedicated,xonotic-linux64-glx,xonotic-linux64-sdl,xonotic-sdl
 private-dev
-private-etc asound.conf,ca-certificates,drirc,fonts,group,host.conf,hostname,hosts,ld.so.cache,ld.so.preload,localtime,nsswitch.conf,passwd,pulse,resolv.conf,ssl,pki,crypto-policies,machine-id
+private-etc asound.conf,ca-certificates,drirc,fonts,group,host.conf,hostname,hosts,ld.so.cache,ld.so.preload,localtime,nsswitch.conf,passwd,pulse,resolv.conf,ssl,pki,crypto-policies,machine-id,alternatives
 private-tmp
 
 noexec ${HOME}
--- firejail-0.9.58.orig/etc/xpra.profile
+++ firejail-0.9.58/etc/xpra.profile
@@ -52,5 +52,5 @@ shell none
 # older Xpra versions also use Xvfb
 # private-bin xpra,python*,Xvfb,Xorg,sh,xkbcomp,xauth,dbus-launch,pactl,ldconfig,which,strace,bash,cat,ls
 private-dev
-# private-etc ld.so.conf,ld.so.cache,resolv.conf,host.conf,nsswitch.conf,gai.conf,hosts,hostname,machine-id,xpra,X11
+# private-etc ld.so.conf,ld.so.cache,resolv.conf,host.conf,nsswitch.conf,gai.conf,hosts,hostname,machine-id,xpra,X11,alternatives
 private-tmp
--- firejail-0.9.58.orig/etc/xreader.profile
+++ firejail-0.9.58/etc/xreader.profile
@@ -38,7 +38,7 @@ tracelog
 
 private-bin xreader,xreader-previewer,xreader-thumbnailer
 private-dev
-private-etc fonts,ld.so.cache
+private-etc fonts,ld.so.cache,alternatives
 private-tmp
 
 memory-deny-write-execute
--- firejail-0.9.58.orig/etc/zathura.profile
+++ firejail-0.9.58/etc/zathura.profile
@@ -35,7 +35,7 @@ shell none
 private-bin zathura
 private-cache
 private-dev
-private-etc fonts,machine-id
+private-etc fonts,machine-id,alternatives
 private-tmp
 
 read-only ${HOME}/
--- firejail-0.9.58.orig/src/firecfg/firecfg.config
+++ firejail-0.9.58/src/firecfg/firecfg.config
@@ -30,9 +30,9 @@ anydesk
 apktool
 arch-audit
 archaudit-report
-ardour4
-ardour5
-arduino
+#ardour4
+#ardour5
+#arduino
 ark
 arm
 artha
@@ -54,7 +54,7 @@ basilisk
 beaker
 bibletime
 bitlbee
-bleachbit
+#bleachbit
 blender
 blender-2.8
 bless
@@ -80,10 +80,10 @@ chromium
 chromium-browser
 cin
 cinelerra
-clamdscan
-clamdtop
-clamscan
-clamtk
+#clamdscan
+#clamdtop
+#clamscan
+#clamtk
 claws-mail
 clawsker
 clementine
@@ -257,20 +257,20 @@ ktorrent
 kwrite
 leafpad
 less
-libreoffice
+#libreoffice
 liferea
 linphone
 lmms
-lobase
-localc
-lodraw
-loffice
-lofromtemplate
-loimpress
+#lobase
+#localc
+#lodraw
+#loffice
+#lofromtemplate
+#loimpress
 lollypop
-lomath
-loweb
-lowriter
+#lomath
+#loweb
+#lowriter
 luminance-hdr
 lximage-qt
 lxmusic
@@ -392,7 +392,7 @@ slack
 smplayer
 smtube
 sol
-soffice
+#soffice
 soundconverter
 spotify
 sqlitebrowser
@@ -455,6 +455,7 @@ vivaldi-beta
 vivaldi-snapshot
 vivaldi-stable
 vlc
+vscodium
 vym
 w3m
 warzone2100
